### ISSUE TITLE
Better explain values returned by useMutation hook

### ISSUE THREAD

**Issue by michaeljwilliams** [FormidableLabs/urql#260]
From docs on useMutation hook:
```js
const [res, executeMutation] = useMutation(query)
```

I'm having some difficulty understanding the difference between `res` and `executeMutation`.

From my testing I see that on a successful mutation `res` resolves to 
```js
{
  data: undefined,
  error: undefined,
  fetching: false,
}
```
and `executeMutation` resolves to
```js
{
  data: {
    __typename: "Mutation",
    myMutationName: {
      ...
    }
  },
  error: undefined,
  operation: {
    ...
  }
}
```
So `executeMutation` has the data I want, but `res` knows when it's done being fetched. Am I supposed to use both `res` and `executeMutation` to handle the query? That's what I'm doing now, but it seems to me like only one should be necessary. And which one would an error be on?

**Comment by kitten** [comment]
The first return value I the tuple is essentially just a convenience, not a necessity. It also makes it more similar to `useQuery`.

Essentially as you found out when triggering the mutation the promise you'll get resolves to the same result. You get the same data and potential error value. So since it's a promise you already know when it's done.

Now the first argument is pretty useful since some mutations you'd want to treat like a fire and forget. You'll want to trigger the mutation but maybe you don't want to wait for the result and save the state yourself.

So basically the results are the same but the first is more convenient for quickly showing the result in your component / UI while using the resolved promise value is convenient for any kind of business logic or composition.

Hope that helps 😀

**Comment by michaeljwilliams** [comment]
Okay, now I understand why `fetching` is omitted from `executeMutation`/the second value. But it sounds to me like you're saying the the first value should have `data` on it as well, but in my testing it was undefined.

**Comment by kitten** [comment]
Yes it'll initially be undefined. It doesn't make much sense to trigger the mutation automatically which is why the hook also doesn't accept variables.

You'd want to trigger a hook manually after a user interaction or another effect typically, so what we do is, we couple the `executeMutation` function to this state.

So when you execute it you won't have to wait for the promise. Instead the first value will trigger updates and rerender your components as the mutation progresses

**Comment by michaeljwilliams** [comment]
I'm seeing `data` undefined on the first value of the tuple even after the second value's promise resolves.
